pkg/apis/openebs.io/scp/v1alpha1/storage-cohort.go

```diff
@@ -27,10 +28,123 @@ import (
 // +kubebuilder:object:root=true
 // +kubebuilder:resource:scope=Namespaced
 type StorageCohort struct {
-	metav1.TypeMeta   `json:",inline"`
+	metav1.TypeMeta `json:",inline"`
+
+	// Standard object's metadata.
 	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the behavior of a cohort.
+	// +required
+	Spec StorageCohortSpec `json:"spec"`
+
+	// Most recently observed status of the cohort.
+	// Populated by the cohort operator or cohort manager.
+	// +optional
+	Status StorageCohortStatus `json:"status,omitempty"`
 }
 
+// StorageCohortSpec describes the attributes that a cohort is created with.
+type StorageCohortSpec struct {
+	// NodeSelector is used to specify the cohort to be considered
+	// +optional
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
+
+	// CohortManager stores all the details about the cohort manager responsible for
+	// managing the cohort
+	// +optional
+	CohortManager interface{} `json:"cohortManager,omitempty"`
+
+	// DefaultStorageProvisioner is the default provisioner for the cohort which can be used
+	// used for provisioning pools or volumes when no provisioner is specified in the storage pool
+	// For example: "openebs.io/scp-lvm-provisioner" or "openebs.io/scp-device-provisioner".
+	// +optional
+	DefaultStorageProvisioner string `json:"defaultStorageProvisioner,omitempty"`
+}
+
+// StorageCohortStatus stores information about the current status of a storage cohort.
+type StorageCohortStatus struct {
+	// Components is an array of different component's conditions which the cohort is comprised of.
+	// +optional
+	Components []ComponentStatus `json:"components,omitempty"`
+
+	// Capabilities represent capabilities that a cohort consists of
+	// +optional
+	Capabilities Capabilities `json:"capabilities,omitempty"`
+}
+
+// ComponentStatus stores information about the current status of storage cohort's components.
+type ComponentStatus struct {
+	// CohortCondition is an array of current observed cohort conditions.
+	// +optional
+	CohortCondition []CohortCondition `json:"cohortCondition,omitempty"`
+
+	// CohortManagerCondition is an array of current observed cohort manager conditions.
+	// +optional
+	CohortManagerCondition []ComponentCondition `json:"cohortManagerCondition,omitempty"`
+
+	// NodeCondition is an array of current observed cohort's individual nodes conditions.
+	// +optional
+	NodeCondition []CohortNodeCondition `json:"nodeCondition,omitempty"`
+}
+
+// CohortNodeCondition contains the latest status information for some or all the
+// nodes that the cohort is comprised of.
+type CohortNodeCondition struct {
+	// Name of the node. This must be a DNS_LABEL.
+	// For example: "virtual-node-1"
+	// +optional
+	Name string `json:"name"`
+
+	// Condition is an array of current observed node conditions.
+	// +optional
+	Condition []ComponentCondition `json:"condition,omitempty"`
```

Comment:
Can we add in the comment what are the different types of node condition

Reply:
These are left upto the user @shovanmaity . You can see that type field is of `string` type which can be defined by the user at their ease.

Reply:
Hmm, If we look at k8s schema all condition type is well defined we need to add define it as it is part of control plane(if we think on overall arch not the mvp) For now it(this pr) is ok.

> These are left upto the user

IMO here user means control plane only.